### PR TITLE
feat(manager/sveltos): add Sveltos ClusterPromotion

### PR DESCRIPTION
**Pull request by gianlucam76**
## Context

Please select one of the following:

- [ ] This closes an existing Issue, Closes: # 
- [x] This doesn't close an Issue, but I accept the risk that this PR may be closed if maintainers disagree with its opening or implementation

## AI assistance disclosure

Did you use AI tools to create any part of this pull request?

Please select one option and, if yes, briefly describe how AI was used (e.g., code, tests, docs) and which tool(s) you used.

- [ ] No — I did not use AI for this contribution.
- [x] Yes — minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [ ] Yes — substantive assistance (AI-generated non‑trivial portions of code, tests, or documentation).
- [ ] Yes — other (please describe):

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests, but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/sveltos/extract.spec.ts [hunk 1]
```diff
@@ -471,6 +501,42 @@ describe('modules/manager/sveltos/extract', () => {
...
+          {
+            currentValue: '0.7.2',
+            datasource: 'docker',
+            depType: 'ClusterPromotion',
+            depName: 'oci://registry-1.docker.io/bitnamicharts/vault',
+            packageName: 'registry-1.docker.io/bitnamicharts/vault',
+          },
+        ],
+      });
+    });
```
Comment (RahulGautamSingh) [inline lib/modules/manager/sveltos/extract.spec.ts:552]: no edge-case test for `ClusterPromotion` without `profileSpec`. the other types have malformed-fixture coverage for null/empty `helmCharts`. Add a `ClusterPromotion` case too.

Comment (gianlucam76) [reply lib/modules/manager/sveltos/extract.spec.ts:552]: Thank you @RahulGautamSingh 

Added empty array and null array for clusterPromotion as well.

File: lib/modules/manager/sveltos/readme.md
Comment (jamietanna) [inline lib/modules/manager/sveltos/readme.md:37]: I realise this is existing docs, but we should potentially remove the duplication here as there's the https://docs.renovatebot.com/modules/manager/sveltos/#dependency-types that gets generated here too

Comment (jamietanna) [reply lib/modules/manager/sveltos/readme.md:37]: Thoughts, Rahul?

Comment (RahulGautamSingh) [reply lib/modules/manager/sveltos/readme.md:37]: Yes, we should remove the duplication. But, I like the external links in the `readme.md` we should keep those.
 
Maybe we can link them through the description fields

**Review by jamietanna — APPROVED** [review]
